chore: Implement Release Please GitHub actions (#1058)
Amsterdam/design-system: documentation/publishing.md

# Publishing

> We use a Release Please GitHub action to create changelog and release PRs for all our packages.

Reviewer: ```suggestion
We use a [Release Please](https://github.com/googleapis/release-please) GitHub action to create a changelog and release PRs for all our packages.
```

Author: Good idea, but I'll link directly to the action itself

> When the release PR is merged, that same action publishes the new release to npm, GitHub releases and deploys it to our main Storybook instance.

Reviewer: ```suggestion
When the release PR is merged, that same action publishes the new release to npm, creates a release on GitHub, and deploys it to our Storybook environment.
```

Author: Hm, we have several Storybook environments though. The reason I wrote 'main' is to clarify that it goes to the main one

> Only the [maintainers](./documentation/maintainers.md) can release new versions of our packages.

Reviewer: ```suggestion
The [maintainers](./documentation/maintainers.md) can release new versions of our packages.
```

## How to create a release

> 1. Locally merge the latest version of `develop` into `main` using a fast-forward merge and push to remote:

Reviewer: ```suggestion
1. Locally merge the latest version of `develop` into `main` using a fast-forward merge, and push to the remote:
```

   ```shell
     git checkout main
     git pull
     git merge --ff-only origin/develop
     git push
   ```

> 2. This triggers an action which creates a release PR.
>    Check this PR and merge it into `main`.
>    This triggers the same action, to publish the release to npm and GitHub releases.
>    It also deploys the released version to our main Storybook instance.

Reviewer: ```suggestion
2. This triggers a GitHub Action, which creates a release PR.
   Review, approve and merge this PR.
   The same Action will then publish the release to npm and GitHub.
   It also deploys the released version to our Storybook environment.
```

> 3. When the release actions have completed, there’s a new release commit on `main`.
>    Locally merge this commit back into `develop` and push to remote:

Reviewer: ```suggestion
3. When complete, the Action adds a new release commit to `main`.
   Locally merge this commit back into `develop` and push it to the remote:
```
